### ISSUE TITLE
Unable to install pycaret on Ubuntu OS!

### ISSUE THREAD

**Issue by mingjun1120** [pycaret/pycaret#1187]
**Failed at installing spacy**
Currently, I am using a PC with Ubuntu OS. I try to type pip install in the terminal to install pycaret but then it fails when the installation reaches Collecting spacy<2.4.0 (from pycaret)

![image](https://user-images.githubusercontent.com/54136320/115617519-1025e080-a324-11eb-83e1-8fa196539040.png)

**Comment by mohsin-ashraf** [comment]
I am also getting error while installing the pycaret[full] (`pip install pycaret` works fine). Can anyone help me what's going wrong
Attaching the logs below

```Collecting slicer==0.0.7
  Using cached slicer-0.0.7-py3-none-any.whl (14 kB)
Requirement already satisfied: patsy>=0.5 in ./.local/lib/python3.7/site-packages (from statsmodels->pyod->pycaret[full]) (0.5.1)
Requirement already satisfied: pynndescent>=0.5 in ./.local/lib/python3.7/site-packages (from umap-learn->pycaret[full]) (0.5.2)
Building wheels for collected packages: gpustat, nvidia-ml-py3, pyperclip, shap
  Building wheel for gpustat (setup.py) ... done
  Created wheel for gpustat: filename=gpustat-0.6.0-py3-none-any.whl size=12617 sha256=0d63b880002e1e1256026939765cc6a113836fe92b2acbcaad909ef8de1e91e8
  Stored in directory: /home/mohsin/.cache/pip/wheels/e6/67/af/f1ad15974b8fd95f59a63dbf854483ebe5c7a46a93930798b8
  Building wheel for nvidia-ml-py3 (setup.py) ... done
  Created wheel for nvidia-ml-py3: filename=nvidia_ml_py3-7.352.0-py3-none-any.whl size=19191 sha256=ac5c0c4ea9356c4288f1b22b2044e91463f8e4557558f97e502418458f2d0198
  Stored in directory: /home/mohsin/.cache/pip/wheels/df/99/da/c34f202dc8fd1dffd35e0ecf1a7d7f8374ca05fbcbaf974b83
  Building wheel for pyperclip (setup.py) ... done
  Created wheel for pyperclip: filename=pyperclip-1.8.2-py3-none-any.whl size=11107 sha256=7dd908410acb2b2b870e99451edb96937260a2c6d041657a5422cb8fa48dc5e8
  Stored in directory: /home/mohsin/.cache/pip/wheels/9f/18/84/8f69f8b08169c7bae2dde6bd7daf0c19fca8c8e500ee620a28
  Building wheel for shap (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3.7 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-umrv5f57/shap_30d8b29e748f4bed896abb8156685f79/setup.py'"'"'; __file__='"'"'/tmp/pip-install-umrv5f57/shap_30d8b29e748f4bed896abb8156685f79/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-lk0girwk
       cwd: /tmp/pip-install-umrv5f57/shap_30d8b29e748f4bed896abb8156685f79/
  Complete output (122 lines):
  The nvcc binary could not be located in your $PATH. Either  add it to your path, or set $CUDAHOME to enable CUDA
  Error building cuda module: TypeError('cannot unpack non-iterable NoneType object')
  WARNING: Could not compile cuda extensions
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.7
  creating build/lib.linux-x86_64-3.7/shap
  copying shap/_serializable.py -> build/lib.linux-x86_64-3.7/shap
  copying shap/_explanation.py -> build/lib.linux-x86_64-3.7/shap
  copying shap/__init__.py -> build/lib.linux-x86_64-3.7/shap
  copying shap/datasets.py -> build/lib.linux-x86_64-3.7/shap
  copying shap/links.py -> build/lib.linux-x86_64-3.7/shap
  creating build/lib.linux-x86_64-3.7/shap/explainers
  copying shap/explainers/pytree.py -> build/lib.linux-x86_64-3.7/shap/explainers
  copying shap/explainers/_permutation.py -> build/lib.linux-x86_64-3.7/shap/explainers
  copying shap/explainers/_gpu_tree.py -> build/lib.linux-x86_64-3.7/shap/explainers
  copying shap/explainers/_partition.py -> build/lib.linux-x86_64-3.7/shap/explainers
  copying shap/explainers/tf_utils.py -> build/lib.linux-x86_64-3.7/shap/explainers
  copying shap/explainers/_kernel.py -> build/lib.linux-x86_64-3.7/shap/explainers
  copying shap/explainers/_exact.py -> build/lib.linux-x86_64-3.7/shap/explainers
  copying shap/explainers/__init__.py -> build/lib.linux-x86_64-3.7/shap/explainers
  copying shap/explainers/mimic.py -> build/lib.linux-x86_64-3.7/shap/explainers
  copying shap/explainers/_linear.py -> build/lib.linux-x86_64-3.7/shap/explainers
  copying shap/explainers/_explainer.py -> build/lib.linux-x86_64-3.7/shap/explainers
  copying shap/explainers/_sampling.py -> build/lib.linux-x86_64-3.7/shap/explainers
  copying shap/explainers/_gradient.py -> build/lib.linux-x86_64-3.7/shap/explainers
  copying shap/explainers/_tree.py -> build/lib.linux-x86_64-3.7/shap/explainers
  copying shap/explainers/_additive.py -> build/lib.linux-x86_64-3.7/shap/explainers
  creating build/lib.linux-x86_64-3.7/shap/explainers/other
  copying shap/explainers/other/_lime.py -> build/lib.linux-x86_64-3.7/shap/explainers/other
  copying shap/explainers/other/_treegain.py -> build/lib.linux-x86_64-3.7/shap/explainers/other
  copying shap/explainers/other/_maple.py -> build/lib.linux-x86_64-3.7/shap/explainers/other
  copying shap/explainers/other/__init__.py -> build/lib.linux-x86_64-3.7/shap/explainers/other
  copying shap/explainers/other/_coefficent.py -> build/lib.linux-x86_64-3.7/shap/explainers/other
  copying shap/explainers/other/_random.py -> build/lib.linux-x86_64-3.7/shap/explainers/other
  creating build/lib.linux-x86_64-3.7/shap/explainers/_deep
  copying shap/explainers/_deep/deep_tf.py -> build/lib.linux-x86_64-3.7/shap/explainers/_deep
  copying shap/explainers/_deep/__init__.py -> build/lib.linux-x86_64-3.7/shap/explainers/_deep
  copying shap/explainers/_deep/deep_pytorch.py -> build/lib.linux-x86_64-3.7/shap/explainers/_deep
  creating build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/_waterfall.py -> build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/_scatter.py -> build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/_embedding.py -> build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/_utils.py -> build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/_partial_dependence.py -> build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/_labels.py -> build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/_violin.py -> build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/_group_difference.py -> build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/_text.py -> build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/__init__.py -> build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/_monitoring.py -> build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/_force_matplotlib.py -> build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/_bar.py -> build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/_heatmap.py -> build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/_force.py -> build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/_decision.py -> build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/_image.py -> build/lib.linux-x86_64-3.7/shap/plots
  copying shap/plots/_beeswarm.py -> build/lib.linux-x86_64-3.7/shap/plots
  creating build/lib.linux-x86_64-3.7/shap/plots/colors
  copying shap/plots/colors/_colorconv.py -> build/lib.linux-x86_64-3.7/shap/plots/colors
  copying shap/plots/colors/_colors.py -> build/lib.linux-x86_64-3.7/shap/plots/colors
  copying shap/plots/colors/__init__.py -> build/lib.linux-x86_64-3.7/shap/plots/colors
  creating build/lib.linux-x86_64-3.7/shap/benchmark
  copying shap/benchmark/methods.py -> build/lib.linux-x86_64-3.7/shap/benchmark
  copying shap/benchmark/perturbation.py -> build/lib.linux-x86_64-3.7/shap/benchmark
  copying shap/benchmark/models.py -> build/lib.linux-x86_64-3.7/shap/benchmark
  copying shap/benchmark/framework.py -> build/lib.linux-x86_64-3.7/shap/benchmark
  copying shap/benchmark/experiments.py -> build/lib.linux-x86_64-3.7/shap/benchmark
  copying shap/benchmark/__init__.py -> build/lib.linux-x86_64-3.7/shap/benchmark
  copying shap/benchmark/measures.py -> build/lib.linux-x86_64-3.7/shap/benchmark
  copying shap/benchmark/metrics.py -> build/lib.linux-x86_64-3.7/shap/benchmark
  copying shap/benchmark/plots.py -> build/lib.linux-x86_64-3.7/shap/benchmark
  creating build/lib.linux-x86_64-3.7/shap/maskers
  copying shap/maskers/_composite.py -> build/lib.linux-x86_64-3.7/shap/maskers
  copying shap/maskers/_output_composite.py -> build/lib.linux-x86_64-3.7/shap/maskers
  copying shap/maskers/_fixed.py -> build/lib.linux-x86_64-3.7/shap/maskers
  copying shap/maskers/_text.py -> build/lib.linux-x86_64-3.7/shap/maskers
  copying shap/maskers/__init__.py -> build/lib.linux-x86_64-3.7/shap/maskers
  copying shap/maskers/_fixed_composite.py -> build/lib.linux-x86_64-3.7/shap/maskers
  copying shap/maskers/_masker.py -> build/lib.linux-x86_64-3.7/shap/maskers
  copying shap/maskers/_tabular.py -> build/lib.linux-x86_64-3.7/shap/maskers
  copying shap/maskers/_image.py -> build/lib.linux-x86_64-3.7/shap/maskers
  creating build/lib.linux-x86_64-3.7/shap/utils
  copying shap/utils/_general.py -> build/lib.linux-x86_64-3.7/shap/utils
  copying shap/utils/_legacy.py -> build/lib.linux-x86_64-3.7/shap/utils
  copying shap/utils/_clustering.py -> build/lib.linux-x86_64-3.7/shap/utils
  copying shap/utils/_show_progress.py -> build/lib.linux-x86_64-3.7/shap/utils
  copying shap/utils/transformers.py -> build/lib.linux-x86_64-3.7/shap/utils
  copying shap/utils/image.py -> build/lib.linux-x86_64-3.7/shap/utils
  copying shap/utils/_keras.py -> build/lib.linux-x86_64-3.7/shap/utils
  copying shap/utils/__init__.py -> build/lib.linux-x86_64-3.7/shap/utils
  copying shap/utils/_masked_model.py -> build/lib.linux-x86_64-3.7/shap/utils
  creating build/lib.linux-x86_64-3.7/shap/actions
  copying shap/actions/_action.py -> build/lib.linux-x86_64-3.7/shap/actions
  copying shap/actions/__init__.py -> build/lib.linux-x86_64-3.7/shap/actions
  copying shap/actions/_optimizer.py -> build/lib.linux-x86_64-3.7/shap/actions
  creating build/lib.linux-x86_64-3.7/shap/models
  copying shap/models/_topk_lm.py -> build/lib.linux-x86_64-3.7/shap/models
  copying shap/models/_text_generation.py -> build/lib.linux-x86_64-3.7/shap/models
  copying shap/models/_teacher_forcing.py -> build/lib.linux-x86_64-3.7/shap/models
  copying shap/models/__init__.py -> build/lib.linux-x86_64-3.7/shap/models
  copying shap/models/_model.py -> build/lib.linux-x86_64-3.7/shap/models
  copying shap/models/_transformers_pipeline.py -> build/lib.linux-x86_64-3.7/shap/models
  creating build/lib.linux-x86_64-3.7/shap/plots/resources
  copying shap/plots/resources/logoSmallGray.png -> build/lib.linux-x86_64-3.7/shap/plots/resources
  copying shap/plots/resources/bundle.js -> build/lib.linux-x86_64-3.7/shap/plots/resources
  creating build/lib.linux-x86_64-3.7/shap/cext
  copying shap/cext/tree_shap.h -> build/lib.linux-x86_64-3.7/shap/cext
  running build_ext
  numpy.get_include() /home/mohsin/.local/lib/python3.7/site-packages/numpy/core/include
  building 'shap._cext' extension
  creating build/temp.linux-x86_64-3.7
  creating build/temp.linux-x86_64-3.7/shap
  creating build/temp.linux-x86_64-3.7/shap/cext
  x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.7m -I/home/mohsin/.local/lib/python3.7/site-packages/numpy/core/include -c shap/cext/_cext.cc -o build/temp.linux-x86_64-3.7/shap/cext/_cext.o
  shap/cext/_cext.cc:3:10: fatal error: Python.h: No such file or directory
   #include <Python.h>
            ^~~~~~~~~~
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for shap
  Running setup.py clean for shap
Successfully built gpustat nvidia-ml-py3 pyperclip
Failed to build shap
Installing collected packages: multidict, yarl, googleapis-common-protos, async-timeout, psutil, opencensus-context, nvidia-ml-py3, hiredis, google-api-core, blessings, aiohttp, redis, pyperclip, py-spy, pbr, opencensus, msgpack, gpustat, aioredis, aiohttp-cors, tensorboardX, stevedore, ray, PrettyTable, isodate, google-crc32c, cmd2, slicer, pyaml, msrest, graphviz, google-resumable-media, google-cloud-core, colorlog, cmaes, cliff, azure-core, xgboost, tune-sklearn, shap, scikit-optimize, optuna, hyperopt, google-cloud-storage, catboost, azure-storage-blob
    Running setup.py install for shap ... error
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3.7 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-umrv5f57/shap_30d8b29e748f4bed896abb8156685f79/setup.py'"'"'; __file__='"'"'/tmp/pip-install-umrv5f57/shap_30d8b29e748f4bed896abb8156685f79/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-mxx2l0ui/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/mohsin/.local/include/python3.7m/shap
         cwd: /tmp/pip-install-umrv5f57/shap_30d8b29e748f4bed896abb8156685f79/
    Complete output (122 lines):
    The nvcc binary could not be located in your $PATH. Either  add it to your path, or set $CUDAHOME to enable CUDA
    Error building cuda module: TypeError('cannot unpack non-iterable NoneType object')
    WARNING: Could not compile cuda extensions
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.7
    creating build/lib.linux-x86_64-3.7/shap
    copying shap/_serializable.py -> build/lib.linux-x86_64-3.7/shap
    copying shap/_explanation.py -> build/lib.linux-x86_64-3.7/shap
    copying shap/__init__.py -> build/lib.linux-x86_64-3.7/shap
    copying shap/datasets.py -> build/lib.linux-x86_64-3.7/shap
    copying shap/links.py -> build/lib.linux-x86_64-3.7/shap
    creating build/lib.linux-x86_64-3.7/shap/explainers
    copying shap/explainers/pytree.py -> build/lib.linux-x86_64-3.7/shap/explainers
    copying shap/explainers/_permutation.py -> build/lib.linux-x86_64-3.7/shap/explainers
    copying shap/explainers/_gpu_tree.py -> build/lib.linux-x86_64-3.7/shap/explainers
    copying shap/explainers/_partition.py -> build/lib.linux-x86_64-3.7/shap/explainers
    copying shap/explainers/tf_utils.py -> build/lib.linux-x86_64-3.7/shap/explainers
    copying shap/explainers/_kernel.py -> build/lib.linux-x86_64-3.7/shap/explainers
    copying shap/explainers/_exact.py -> build/lib.linux-x86_64-3.7/shap/explainers
    copying shap/explainers/__init__.py -> build/lib.linux-x86_64-3.7/shap/explainers
    copying shap/explainers/mimic.py -> build/lib.linux-x86_64-3.7/shap/explainers
    copying shap/explainers/_linear.py -> build/lib.linux-x86_64-3.7/shap/explainers
    copying shap/explainers/_explainer.py -> build/lib.linux-x86_64-3.7/shap/explainers
    copying shap/explainers/_sampling.py -> build/lib.linux-x86_64-3.7/shap/explainers
    copying shap/explainers/_gradient.py -> build/lib.linux-x86_64-3.7/shap/explainers
    copying shap/explainers/_tree.py -> build/lib.linux-x86_64-3.7/shap/explainers
    copying shap/explainers/_additive.py -> build/lib.linux-x86_64-3.7/shap/explainers
    creating build/lib.linux-x86_64-3.7/shap/explainers/other
    copying shap/explainers/other/_lime.py -> build/lib.linux-x86_64-3.7/shap/explainers/other
    copying shap/explainers/other/_treegain.py -> build/lib.linux-x86_64-3.7/shap/explainers/other
    copying shap/explainers/other/_maple.py -> build/lib.linux-x86_64-3.7/shap/explainers/other
    copying shap/explainers/other/__init__.py -> build/lib.linux-x86_64-3.7/shap/explainers/other
    copying shap/explainers/other/_coefficent.py -> build/lib.linux-x86_64-3.7/shap/explainers/other
    copying shap/explainers/other/_random.py -> build/lib.linux-x86_64-3.7/shap/explainers/other
    creating build/lib.linux-x86_64-3.7/shap/explainers/_deep
    copying shap/explainers/_deep/deep_tf.py -> build/lib.linux-x86_64-3.7/shap/explainers/_deep
    copying shap/explainers/_deep/__init__.py -> build/lib.linux-x86_64-3.7/shap/explainers/_deep
    copying shap/explainers/_deep/deep_pytorch.py -> build/lib.linux-x86_64-3.7/shap/explainers/_deep
    creating build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/_waterfall.py -> build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/_scatter.py -> build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/_embedding.py -> build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/_utils.py -> build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/_partial_dependence.py -> build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/_labels.py -> build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/_violin.py -> build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/_group_difference.py -> build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/_text.py -> build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/__init__.py -> build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/_monitoring.py -> build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/_force_matplotlib.py -> build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/_bar.py -> build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/_heatmap.py -> build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/_force.py -> build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/_decision.py -> build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/_image.py -> build/lib.linux-x86_64-3.7/shap/plots
    copying shap/plots/_beeswarm.py -> build/lib.linux-x86_64-3.7/shap/plots
    creating build/lib.linux-x86_64-3.7/shap/plots/colors
    copying shap/plots/colors/_colorconv.py -> build/lib.linux-x86_64-3.7/shap/plots/colors
    copying shap/plots/colors/_colors.py -> build/lib.linux-x86_64-3.7/shap/plots/colors
    copying shap/plots/colors/__init__.py -> build/lib.linux-x86_64-3.7/shap/plots/colors
    creating build/lib.linux-x86_64-3.7/shap/benchmark
    copying shap/benchmark/methods.py -> build/lib.linux-x86_64-3.7/shap/benchmark
    copying shap/benchmark/perturbation.py -> build/lib.linux-x86_64-3.7/shap/benchmark
    copying shap/benchmark/models.py -> build/lib.linux-x86_64-3.7/shap/benchmark
    copying shap/benchmark/framework.py -> build/lib.linux-x86_64-3.7/shap/benchmark
    copying shap/benchmark/experiments.py -> build/lib.linux-x86_64-3.7/shap/benchmark
    copying shap/benchmark/__init__.py -> build/lib.linux-x86_64-3.7/shap/benchmark
    copying shap/benchmark/measures.py -> build/lib.linux-x86_64-3.7/shap/benchmark
    copying shap/benchmark/metrics.py -> build/lib.linux-x86_64-3.7/shap/benchmark
    copying shap/benchmark/plots.py -> build/lib.linux-x86_64-3.7/shap/benchmark
    creating build/lib.linux-x86_64-3.7/shap/maskers
    copying shap/maskers/_composite.py -> build/lib.linux-x86_64-3.7/shap/maskers
    copying shap/maskers/_output_composite.py -> build/lib.linux-x86_64-3.7/shap/maskers
    copying shap/maskers/_fixed.py -> build/lib.linux-x86_64-3.7/shap/maskers
    copying shap/maskers/_text.py -> build/lib.linux-x86_64-3.7/shap/maskers
    copying shap/maskers/__init__.py -> build/lib.linux-x86_64-3.7/shap/maskers
    copying shap/maskers/_fixed_composite.py -> build/lib.linux-x86_64-3.7/shap/maskers
    copying shap/maskers/_masker.py -> build/lib.linux-x86_64-3.7/shap/maskers
    copying shap/maskers/_tabular.py -> build/lib.linux-x86_64-3.7/shap/maskers
    copying shap/maskers/_image.py -> build/lib.linux-x86_64-3.7/shap/maskers
    creating build/lib.linux-x86_64-3.7/shap/utils
    copying shap/utils/_general.py -> build/lib.linux-x86_64-3.7/shap/utils
    copying shap/utils/_legacy.py -> build/lib.linux-x86_64-3.7/shap/utils
    copying shap/utils/_clustering.py -> build/lib.linux-x86_64-3.7/shap/utils
    copying shap/utils/_show_progress.py -> build/lib.linux-x86_64-3.7/shap/utils
    copying shap/utils/transformers.py -> build/lib.linux-x86_64-3.7/shap/utils
    copying shap/utils/image.py -> build/lib.linux-x86_64-3.7/shap/utils
    copying shap/utils/_keras.py -> build/lib.linux-x86_64-3.7/shap/utils
    copying shap/utils/__init__.py -> build/lib.linux-x86_64-3.7/shap/utils
    copying shap/utils/_masked_model.py -> build/lib.linux-x86_64-3.7/shap/utils
    creating build/lib.linux-x86_64-3.7/shap/actions
    copying shap/actions/_action.py -> build/lib.linux-x86_64-3.7/shap/actions
    copying shap/actions/__init__.py -> build/lib.linux-x86_64-3.7/shap/actions
    copying shap/actions/_optimizer.py -> build/lib.linux-x86_64-3.7/shap/actions
    creating build/lib.linux-x86_64-3.7/shap/models
    copying shap/models/_topk_lm.py -> build/lib.linux-x86_64-3.7/shap/models
    copying shap/models/_text_generation.py -> build/lib.linux-x86_64-3.7/shap/models
    copying shap/models/_teacher_forcing.py -> build/lib.linux-x86_64-3.7/shap/models
    copying shap/models/__init__.py -> build/lib.linux-x86_64-3.7/shap/models
    copying shap/models/_model.py -> build/lib.linux-x86_64-3.7/shap/models
    copying shap/models/_transformers_pipeline.py -> build/lib.linux-x86_64-3.7/shap/models
    creating build/lib.linux-x86_64-3.7/shap/plots/resources
    copying shap/plots/resources/logoSmallGray.png -> build/lib.linux-x86_64-3.7/shap/plots/resources
    copying shap/plots/resources/bundle.js -> build/lib.linux-x86_64-3.7/shap/plots/resources
    creating build/lib.linux-x86_64-3.7/shap/cext
    copying shap/cext/tree_shap.h -> build/lib.linux-x86_64-3.7/shap/cext
    running build_ext
    numpy.get_include() /home/mohsin/.local/lib/python3.7/site-packages/numpy/core/include
    building 'shap._cext' extension
    creating build/temp.linux-x86_64-3.7
    creating build/temp.linux-x86_64-3.7/shap
    creating build/temp.linux-x86_64-3.7/shap/cext
    x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.7m -I/home/mohsin/.local/lib/python3.7/site-packages/numpy/core/include -c shap/cext/_cext.cc -o build/temp.linux-x86_64-3.7/shap/cext/_cext.o
    shap/cext/_cext.cc:3:10: fatal error: Python.h: No such file or directory
     #include <Python.h>
              ^~~~~~~~~~
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python3.7 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-umrv5f57/shap_30d8b29e748f4bed896abb8156685f79/setup.py'"'"'; __file__='"'"'/tmp/pip-install-umrv5f57/shap_30d8b29e748f4bed896abb8156685f79/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-mxx2l0ui/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/mohsin/.local/include/python3.7m/shap Check the logs for full command output.
```

**Comment by pycaret** [comment]
@mohsin-ashraf The error you get with `pip install pycaret[full]` is due to `shap` library. 

Try installing shap with this command and then run `pip install pycaret[full]`:

```
conda install -c conda-forge shap
```

**Comment by mingjun1120** [comment]
@pycaret How about me?

**Comment by pycaret** [comment]
@mingjun1120 You are getting the error because of spacy. Try this before installing pycaret:

```
conda install -c conda-forge spacy
```

**Comment by mohsin-ashraf** [comment]
@mingjun1120 if you still face the problem I would suggest you going to `python3.6` since The suggestion by @pycaret did not work on `python3.7` then I switched to python3.6 and it worked!

**Comment by pycaret** [comment]
@mingjun1120 Is your issue resolved with `python==3.6`

**Comment by mingjun1120** [comment]
> @mingjun1120 Is your issue resolved with `python==3.6`

No. I also don't know why. I don't use the Ubuntu OS PC anymore as it is my uni's lab PC.
